### PR TITLE
Fixed #5908 - SMTP Server not configured after upgrading to 7.10.5

### LINKED ISSUE #5908: SMTP Server not configured after upgrading to 7.10.5
#### Issue

After upgrading to 7.10.5 I see this note whenever I enter the crm as admin:
**Note: To send record assignment notifications, an SMTP server must be configured in Email Settings.**

If I click on the Email Settings link, I find empty SMTP Server fields. After filling data and saving, the Note is still there:
schermata 2018-05-21 alle 18 23 33

If I click again, the SMTP fields are still empty:
schermata 2018-05-21 alle 18 50 40

If I try to send an email, it is sent without any error (just like the SMTP Server credentials were there).

I checked the outbound_table with phpMyAdmin, and the record is actually there.

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

Fixes an issue that was causing the SMTP settings to continue instead of saving.

Issue reference: #5908

## How To Test This

1. Go into admin panel -> email settings
2. Save SMTP settings

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: modules/Administration/Administration.php [hunk 1]
```diff
@@ -172,7 +164,7 @@ public function retrieveSettings($category = false, $clean = false)
 
             foreach ($oe->field_defs as $def) {
                 // fixes installer php notice
-                if (!array_key_exists($def, $this->settings)) {
+                if (!isset($def) && (!isset($this->settings))) {
```
Comment (zoltankocsardi) [inline modules/Administration/Administration.php]: It's not necessary using brackets before the second isset.

**Comment by Kristianostman** [comment]
This fix is not working.

**Comment by PedroErnst** [comment]
Tested and works for me.

**Comment by samus-aran** [comment]
Tests are failing due to that PHP Notice. Update PR to allow this to pass :ok_hand:

**Comment by samus-aran** [comment]
@Kristianostman 
Can you please provide more information on how you tested, what you tested with and how does the failure manifests. Thanks

**Comment by Kristianostman** [comment]
@samus-aran I will recheck again as soon as possible. I did according to above fix instructions. Since it did not work for me I rolled back. Though, it may be that my problems are coming from cache related issues on the server. I have recognised earlier that code changes require some time to show up not enough with emptying browser cache and run the repair / rebuild.

**Comment by daniel-samson** [comment]
Assessed

**Pull request merged**